gen/string.go

```diff
@@ -70,12 +74,14 @@ func pascalCase(words ...string) string {
 	return strings.Join(words, "")
 }
 
+func constantName(s string) string {
+	return pascalCase(false /* all caps */, strings.Split(s, "_")...)
+}
+
 // enumItemName returns the Go name that should be used for an enum item with
 // the given Thrift name.
 func enumItemName(enumName string, itemName string) string {
-	words := []string{enumName}
-	words = append(words, strings.Split(itemName, "_")...)
-	return pascalCase(words...)
+	return enumName + pascalCase(false /* all caps */, strings.Split(itemName, "_")...)
```

Comment:
Actually, I'm also going to change `enumItemName` to accept the `EnumSpec` and generate the name from that so that there's no change of accidentally calling it with the wrong enum name.